core/src/main/java/org/apache/druid/data/input/impl/JSONParseSpec.java

```diff
@@ -37,46 +37,48 @@
 {
   private final ObjectMapper objectMapper;
   private final Map<String, Boolean> featureSpec;
+  private final Boolean keepNullColumns;
```

Comment:
```suggestion
  private final boolean keepNullColumns;
```

This way the class does not need to worry about nullability issues everywhere, it can just be handled in the constructor.